adagrams/game.py

```diff
@@ -1,11 +1,273 @@
+import random
+
 def draw_letters():
-    pass
+    
+    letter_pool_str= "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    # List comp for converting ABC string on line 7 into list 
+    letter_pool_list = [letter for letter in letter_pool_str]
+    
+    # Original letter pool dict - stays constant
+    letter_pool_dict = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    # Tracking container: where the letters(key) and their freq (value) will be stored as separate dictionaries
+    tracking_letter_dict = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+
+    # List of randomly drawn hand, stores 10 letters or tiles
+    hand = []
+    
+    # Adding a random letter to the hand, and reducing frequency for each draw
+    while len(hand) < 10:
+        random_letter = letter_pool_list[random.randint(0, 25)]
+        initial_letter_count = letter_pool_dict[random_letter]
+        random_letter_count = tracking_letter_dict[random_letter]
+        if 0 < random_letter_count <= initial_letter_count:
+            hand.append(random_letter)
+            tracking_letter_dict[random_letter] -= 1 # *** Located and fixed bug causing one-off error ***
+            
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # Copies of arguments to stay in the "safe" zone:
+    letter_bank_copy = [letter for letter in letter_bank]
+    uppercase_word = word.upper()
+    word_list_copy = [letter for letter in uppercase_word]
```

Comment:
I see you made a `uppercase_word` into a list. However, it seems like it was only looped over and that would be done with `uppercase_word` as well. Using that variable also helps keeps the amount of memory your function is using lower as well.